src/main/java/sirius/web/templates/pdf/handlers/BarcodePdfReplaceHandler.java

```diff
@@ -73,65 +69,40 @@ public FSImage resolveUri(String uri, UserAgentCallback userAgentCallback, int c
         return fsImage;
     }
 
-    private int calculateBarcodeScaleFactor(int cssWidth, int cssHeight, Image awtImage) {
-        return (int) Math.max(Math.ceil(cssWidth / (float) awtImage.getWidth(null)),
-                              Math.ceil(cssHeight / (float) awtImage.getHeight(null)));
-    }
+    private Image generateBarcodeImage(String barcodeType, String content, int width, int height)
+            throws WriterException {
+        assertSupportedBarcodeType(barcodeType);
 
-    /**
-     * Creates an instance of {@link Barcode} that matches the given type descriptor.
-     *
-     * @param type the requested type
-     * @return the barcode
-     */
-    @Nonnull
-    private Barcode createBarcode(String type) {
-        if (BARCODE_TYPE_CODE128.equalsIgnoreCase(type)) {
-            return new Barcode128();
-        }
-
-        if (BARCODE_TYPE_EAN.equalsIgnoreCase(type)) {
-            return new BarcodeEAN();
+        if (BARCODE_TYPE_INTERLEAVED_2_OF_5_CHECKSUMMED.equalsIgnoreCase(barcodeType)) {
+            Barcode code = new BarcodeInter25();
+            code.setGenerateChecksum(true);
+            // Pads the code if necessary: Length is even but a checksum will be added
+            code.setCode(BarcodeInter25.keepNumbers(content).length() % 2 == 0 ? "0" + content : content);
+            return code.createAwtImage(Color.BLACK, Color.WHITE);
```

Comment:
Can this somehow be replaced by zxing?

Reply:
Didn't find a suitable way to do this with zxing.

There were some discussions about this (e.g. https://github.com/zxing/zxing/issues/858), but also no obvious suggested solution.

Reply:
Solved by applying the calculation of the previously used library, which is still used for PDF Images anyway. Barcode rendering itself is now fully performed using zxing.